pkg/webui/lib/host-from-url.js

```diff
@@ -19,8 +19,19 @@
  * @returns {string?} - The hostname of the `url` or undefined.
  */
 export default url => {
+  const IPv6RegexExp =
+    /(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))/gi
+
   try {
-    return new URL(url).hostname
+    if (url.match(IPv6RegexExp)) {
+      return url
+    } else if (url.includes('://')) {
+      return new URL(url).hostname
+    } else if (url.includes(':')) {
+      return new URL(`https://${url}`).hostname
+    }
+
+    return url
   } catch (error) {
```

Comment:
```suggestion
  try {
    if (url.matches(/^[a-zA-Z0-9]+:\/\/.*/)) {
      return new URL(url).hostname
    }
    return new URL(`http://${url}`).hostname
  } catch (error) {
    return url
  }
```

Let's simplify this.